src/camera.c

```diff
@@ -6,43 +6,50 @@
 /*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/10/28 11:48:56 by u413q             #+#    #+#             */
-/*   Updated: 2023/12/22 12:02:39 by gwolf            ###   ########.fr       */
+/*   Updated: 2023/12/25 08:47:19 by gwolf            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "camera.h"
 
-void	ft_initiate_camera(t_cam *cam)
+void	ft_cam_init(t_cam *cam, t_vec2i img_size)
 {
-	cam->vup.x = 0.0;
-	cam->vup.y = 1.0;
-	cam->vup.z = 0.0;
-	cam->camera_centre.x = cam->look_from.x;
-	cam->camera_centre.y = cam->look_from.y;
-	cam->camera_centre.z = cam->look_from.z;
-	cam->w = ft_vec3_norm(ft_vec3_sub(cam->look_from, cam->look_at));
-	cam->u = ft_vec3_cross(cam->vup, cam->w);
+	ft_cam_calc_base_vectors(cam);
+	cam->focal_length = 10.0;
```

Comment:
See before

Reply:
As explained above, that line should be adapted to either

cam->focal_length = ft_vec3_abs(ft_vec3_sub(cam->look_from, cam->look_at));

or even faster

cam->focal_length = ft_vec3_abs(cam->direction);

Reply:
Or just cam->focal_length = 1 since its normalized?

Reply:
See above